packages/presentation/lib/src/pages/movie_details_page/mappers/movie_to_movie_details.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:domain/domain.dart';
+import 'package:presentation/src/pages/movie_details_page/model/movie_details.dart';
+
+class MovieToMovieDetailsMapper implements Mapper<Movie, MovieDetails> {
+  final DurationToStringMapper durationToString;
+  final MovieIdToImageUrlMapper movieIdToImage;
+  final MovieRatingToStarsCountMapper movieRatingToStarsCount;
+  final ListToGenresStringMapper listToGenres;
+
+  const MovieToMovieDetailsMapper({
+    required this.durationToString,
+    required this.movieIdToImage,
+    required this.movieRatingToStarsCount,
+    required this.listToGenres,
+  });
+
+  @override
+  MovieDetails call(Movie movie) => MovieDetails(
+        movie.title.stringOrEmpty,
+        movie.overview.stringOrEmpty,
+        movie.runtime == null
+            ? ''
+            : durationToString(movie.runtime!) +
+                ' ' +
+                movie.certification.stringOrEmpty,
+        movie.genres == null ? '' : listToGenres(movie.genres!),
+        movie.ids?.imdb == null ? '' : movieIdToImage(movie.ids!.imdb!),
+        movie.rating == null ? 0 : movieRatingToStarsCount(movie.rating!),
+        movie.rating == null ? '' : movie.rating!.toStringRating(),
```

Comment:
extract fields which u are checking for null to the separate variables. in this case we can avoid force-unwrap